===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ swAioBase_read:n$1 == { -1 } }) free(*(swAioBase_read:aio_ev)) at 5 (line 373, column 9) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/swoole-src/src/os/base.c
+++ b/table1/benchmarks/swoole-src/src/os/base.c
@@ -370,7 +370,10 @@ static int swAioBase_read(int fd, void *inbuf, size_t size, off_t offset)
     aio_ev->offset = offset;
     aio_ev->task_id = SwooleAIO.current_id++;
 
-    if (swThreadPool_dispatch(&swAioBase_thread_pool, aio_ev, sizeof(aio_ev)) < 0)
+    int tmp = swThreadPool_dispatch(&swAioBase_thread_pool, aio_ev, sizeof(aio_ev));
+    if (tmp == -1)
+      free(aio_ev);
+    if (tmp < 0)
     {
         return SW_ERR;
     }
